### DISCUSSION TITLE
Tab API doesn't pick up editor tabs outside of the main area

### DISCUSSION THREAD

**Discussion by msujew** [eclipse-theia/theia#12157] [Improvements] [answered]
We implemented the VSCode tab API support (https://github.com/eclipse-theia/theia/pull/12109) by examination of the behavior of vscode. While this works perfectly fine when we use Theia like VSCode (with its limited flexibility in regards to widget placement), it starts to break down once we make use of Theia's ability to move any widget to any position in the application:

https://user-images.githubusercontent.com/4377073/217277006-0e199595-54b8-4464-a235-80bd28b5cfb6.mp4

As you can see in the video, widgets outside of the main area don't trigger any events related to the tab API. This is in line with VSCode, but doesn't really make sense in the context of Theia.

The question is now whether and how to deal with this. Placing similar restrictions on Theia doesn't seem to be really reasonable, while extending the tab API to listen to any widget on any panel sounds like overkill for me. I would like to hear/read the opinion of other contributors what they think about this topic, and whether they see any need to "fix" this issue at all.

**Comment by paul-marechal** [comment, marked as answer]
> As you can see in the video, widgets outside of the main area don't trigger any events related to the tab API. This is in line with VSCode, but doesn't really make sense in the context of Theia.

As long as it makes sense with regards to the VS Code API then I think the implementation is fine.

Otherwise extending the VS Code API would mostly mean adding new enum values to `ViewColumn` to identify the extra tab groups that Theia supports?

**Comment by JonasHelming** [comment]
@msujew : I think we can close this as answered, are you fine with this?
